src/huggingface_hub/hf_api.py

```diff
@@ -1094,82 +1127,80 @@ def model_info(
         return ModelInfo(**d)
 
     @_deprecate_positional_args
-    def list_repo_files(
+    def dataset_info(
         self,
         repo_id: str,
         *,
         revision: Optional[str] = None,
-        repo_type: Optional[str] = None,
         token: Optional[str] = None,
         timeout: Optional[float] = None,
-    ) -> List[str]:
+    ) -> DatasetInfo:
         """
-        Get the list of files in a given repo.
+        Get info on one specific dataset on huggingface.co
```

Comment:
```suggestion
        Get info on one specific dataset on huggingface.co.
```

I think the standard requires that the sentence finishes with a dot. I know numpydoc tests for it.